tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java

```diff
@@ -20,78 +20,309 @@
-            new ConsumerGroupListing(simpleGroup, true, Optional.of(ConsumerGroupState.EMPTY)),
-            new ConsumerGroupListing(GROUP, false, Optional.of(ConsumerGroupState.STABLE))));
+        Set<ConsumerGroupListing> expectedListing = new HashSet<>(Arrays.asList(
+            new ConsumerGroupListing(
+                simpleGroup,
+                true,
+                Optional.of(ConsumerGroupState.EMPTY),
+                Optional.of(GroupType.CLASSIC)
+            ),
+            new ConsumerGroupListing(
+                GROUP,
+                false,
+                Optional.of(ConsumerGroupState.STABLE),
+                Optional.of(GroupType.CLASSIC)
+            )
+        ));
 
-        final AtomicReference<scala.collection.Set> foundListing = new AtomicReference<>();
-        TestUtils.waitForCondition(() -> {
-            foundListing.set(service.listConsumerGroupsWithState(set(Arrays.asList(ConsumerGroupState.values()))).toSet());
-            return Objects.equals(expectedListing, foundListing.get());
-        }, "Expected to show groups " + expectedListing + ", but found " + foundListing.get());
+        assertGroupListing(
+            service,
+            Collections.emptySet(),
+            EnumSet.allOf(ConsumerGroupState.class),
+            expectedListing
+        );
 
-        scala.collection.Set<ConsumerGroupListing> expectedListingStable = set(Collections.singleton(
-            new ConsumerGroupListing(GROUP, false, Optional.of(ConsumerGroupState.STABLE))));
+        expectedListing = new HashSet<>(Arrays.asList(
+            new ConsumerGroupListing(
+                GROUP,
+                false,
+                Optional.of(ConsumerGroupState.STABLE),
+                Optional.of(GroupType.CLASSIC)
+            )
+        ));
 
-        foundListing.set(null);
+        assertGroupListing(
+            service,
+            Collections.emptySet(),
+            new HashSet<>(Arrays.asList(ConsumerGroupState.STABLE)),
+            expectedListing
+        );
 
-        TestUtils.waitForCondition(() -> {
-            foundListing.set(service.listConsumerGroupsWithState(set(Collections.singleton(ConsumerGroupState.STABLE))).toSet());
-            return Objects.equals(expectedListingStable, foundListing.get());
-        }, "Expected to show groups " + expectedListingStable + ", but found " + foundListing.get());
+        assertGroupListing(
+            service,
+            Collections.emptySet(),
+            new HashSet<>(Arrays.asList(ConsumerGroupState.PREPARING_REBALANCE)),
+            Collections.emptySet()
+        );
+    }
+
+    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES)
+    @MethodSource("getTestQuorumAndGroupProtocolParametersConsumerGroupProtocolOnly")
+    public void testListConsumerGroupsWithStatesConsumerProtocol(String quorum, String groupProtocol) throws Exception {
```

Comment:
Could we keep `testListConsumerGroupsWithStatesClassicProtocol` and `testListConsumerGroupsWithStatesConsumerProtocol` combined into one? It seems to me that we could derive the type of the second group from `groupProtocol` with `GroupType.parse(groupProtocol)`. Would it work?